src/smpp_socket.erl

```diff
@@ -667,25 +659,54 @@ bind(#{role := Role} = State) ->
     State2 = maps:without([keepalive_time, response_time], State1),
     {next_state, binding, State2, bind_timeout(State2)}.
 
+-spec keep_state(state()) -> gen_statem:event_handler_result(statename()).
+keep_state(State) ->
+    keep_state(State, []).
+
+-spec keep_state(state(), [gen_statem:action()]) ->
```

Comment:
Whouldn't 
```erlang
-spec next_state(statename() | keep_state, state()) ->
                         gen_statem:event_handler_result(statename()).
...
-spec next_state(statename() | keep_state, state(), [gen_statem:action()]) ->
                         gen_statem:event_handler_result(statename()).
...
```
be enough?

Reply:
I'll prefer separate functions for this different purposes

Reply:
removed copypasta